src/utilities/path.ts

```diff
@@ -1,19 +1,37 @@
-import { Param } from '@/types/params'
-import { ExtractParamsFromPathString } from '@/types/routeMethods'
-import { Identity } from '@/types/utilities'
+import { optional, Param, ExtractParamsFromPathString, Identity } from '@/types'
+import { combineParams } from '@/utilities'
 
 type PathParams<T extends string> = {
   [K in keyof ExtractParamsFromPathString<T>]?: Param
 }
 
 export type Path<
   T extends string = any,
-  P extends PathParams<T> = any
+  P extends Record<string, Param[]> = any
 > = {
   path: T,
-  params: P,
+  params: Required<P>,
```

Comment:
I think we could do better here. I'm guessing this is also breaking the route method types which currently depend on Path['Params'] being a non generic value.